packages/flutter_tools/test/general.shard/plugins_test.dart

```diff
@@ -946,6 +945,37 @@ web_plugin_with_nested:${webPluginWithNestedFile.childDirectory('lib').uri.toStr
         FeatureFlags: () => featureFlags,
       });
 
+      testUsingContext('Injecting does not create a generated Windows registrant if no pluginClass is defined', () async {
```

Comment:
Every project needs the generated registrant files. This test is asserting that a project with no plugins won't compile ;)

What you should be testing is that a plugin without a `pluginClass` doesn't show up in the registrant (check that the contents doesn't contain SomePlugin, for instance, for the cc file)